src/vault/use-cases/migrations/eth_to_keys.go

```diff
@@ -0,0 +1,107 @@
+package migrations
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/hex"
+	"github.com/consensys/quorum-hashicorp-vault-plugin/src/pkg/errors"
+	"github.com/consensys/quorum-hashicorp-vault-plugin/src/pkg/log"
+	"github.com/consensys/quorum-hashicorp-vault-plugin/src/vault/entities"
+	"time"
+
+	usecases "github.com/consensys/quorum-hashicorp-vault-plugin/src/vault/use-cases"
+	"github.com/hashicorp/vault/sdk/logical"
+)
+
+type ethToKeysUseCase struct {
+	storage      logical.Storage
+	ethUseCases  usecases.ETHUseCases
+	keysUseCases usecases.KeysUseCases
+	status       map[string]*entities.MigrationStatus
+}
+
+func NewEthToKeysUseCase(ethUseCases usecases.ETHUseCases, keysUseCases usecases.KeysUseCases) usecases.EthereumToKeysUseCase {
+	return &ethToKeysUseCase{
+		ethUseCases:  ethUseCases,
+		keysUseCases: keysUseCases,
+		status:       map[string]*entities.MigrationStatus{},
+	}
+}
+
+func (uc ethToKeysUseCase) WithStorage(storage logical.Storage) usecases.EthereumToKeysUseCase {
+	uc.storage = storage
+	return &uc
+}
+
+func (uc ethToKeysUseCase) Status(ctx context.Context, namespace string) (*entities.MigrationStatus, error) {
+	logger := log.FromContext(ctx).With("namespace", namespace)
+
+	status := uc.status[namespace]
+	if status == nil {
+		errMessage := "migration could not be found"
+		logger.Warn(errMessage)
+		return nil, errors.NotFoundError(errMessage)
+	}
+
+	return status, nil
+}
+
+func (uc *ethToKeysUseCase) Execute(ctx context.Context, namespace string) error {
+	logger := log.FromContext(ctx).With("namespace", namespace)
+
+	addresses, err := uc.ethUseCases.ListAccounts().WithStorage(uc.storage).Execute(ctx, namespace)
+	if err != nil {
+		return err
+	}
+
+	status := &entities.MigrationStatus{
+		Status:    "pending",
+		StartTime: time.Now(),
+	}
+	uc.status[namespace] = status
+
+	go func() {
```

Comment:
IMO making an UC runs in async within go-routine it is a decision to be made on the caller so you might want to run the same UC synchonously for your integration-test for instance or whatever other cases.